tests/unit-tests/test-class-lesson.php

```diff
@@ -101,13 +100,110 @@ public function testIsPreRequisiteComplete() {
 
 	}
 
+	public function testGetCourseId() {
+		$this->assertTrue(
+			method_exists( 'WooThemes_Sensei_Lesson', 'get_course_id' ),
+			'The lesson class method `get_course_id` does not exist '
+		);
+		$course_ids             = $this->factory->course->create_many( 3 );
+		$lesson_ids             = $this->factory->lesson->create_many( 9 );
+		$lesson_id_to_course_id = array();
+		foreach ( $lesson_ids as $lesson_id ) {
+			$course_index                         = array_rand( $course_ids );
```

Comment:
hmmm, good point. I changed it in [29ee76f](https://github.com/Automattic/sensei/pull/5294/commits/29ee76fe5f832d42930fff4fa99992fb3b473a19).

Reply:
Hey there.

> My point of simplifying that is that we could remove the $lesson_id_to_course_id because since we'll know the respective > course, we wouldn't need that, right?

Fair enough. I adapted the code to avoid the usage of this variable and instead compute the expected course ID manually. Please note that this isn't as simple as a `%` operation, because this only returns the expected course index, so we need to get the course ID from the course list anyway. Here are some commits where I basically fix that: [f96947a](https://github.com/Automattic/sensei/pull/5294/commits/f96947a87d5afc503645565b242d1aae7a05d2a5) and [7b2a7c9](https://github.com/Automattic/sensei/pull/5294/commits/7b2a7c930e3c8b43fcbec3a2ce155166a31d5156) (I had to copy the lesson IDs array to avoid issues with the shuffled list, by the way).

> And I think we just forgot the testGetCourseId that still has the array_rand.

I fixed that in [84b345f](https://github.com/Automattic/sensei/pull/5294/commits/84b345f59fcc253f190ddbcc0bb933794088263c). The commit message refers only to the formatting of the PHP file, but it includes that change too.

> Reading a little more the `testGetCourseId`, actually I don't think we need a loop here, right?
> Maybe we could just create one course with a lesson, and check the `get_course_id`?

Actually, kinda. While for content drip specifically usually we always get the same course ID for every lesson queried, this method is meant to be...more generic, so my line of thinking is more like "if it can get a bunch of different courses IDs for different lessons IDs, then it will be able to determine the same course ID for a bunch of lesson IDs".

Does that make sense? Given that the API allows that, testing for that case looks like a better idea, IMHO.

Reply:
aaaahhhh, actually, thinking again, you are right, @renatho. 

I thought you were referring to the `testGetCourseIds` method, to get the course IDs for multiple lessons IDs. I got confused in the name of my own methods. :upside_down_face: 

I just pushed the commit 151b298005e69b65ccaa482a67044792d5211bfd fixing this. :)

Reply:
haha! no problem!